### PR TITLE
Add V4 Axis Properties

### PR DESCRIPTION
**Pull request by melissatdiamond**
Addressing the Axis portion of issue #3779. Initial commit moves properties that are now in Axis in Vega 4 from`VgAxisConfig` to `VgAxisBase`.

### CODE REVIEW

**Comment by domoritz** [comment]
Why do the examples change?

File: src/vega.schema.ts [hunk 1]
```diff
@@ -503,187 +525,187 @@ export type VgBinding = VgCheckboxBinding | VgRadioBinding |
...
    */
-  labels?: boolean;
+  domain?: boolean;
 
   /**
-   * Indicates if labels should be hidden if they exceed the axis range. If `false `(the default) no bounds overlap analysis is performed. If `true`, labels will be hidden if they exceed the axis range by more than 1 pixel. If this property is a number, it specifies the pixel tolerance: the maximum amount by which a label bounding box may exceed the axis range.
+   * Stroke width of axis domain line
    *
-   * __Default value:__ `false`.
+   * __Default value:__  (none, using Vega default).
```
Comment (kanitw) [inline src/vega.schema.ts]: I think saying "(none, using Vega default)" is almost equivalent to not saying anything as users don't really the information they should know. 

You can look at https://github.com/vega/vega-parser/blob/master/src/config.js to see some defaults.  (`domainWidth` is `1`, for example.)

Same for all other properties. :)

File: src/vega.schema.ts [hunk 2]
```diff
@@ -503,187 +525,187 @@ export type VgBinding = VgCheckboxBinding | VgRadioBinding |
...
-   *
-   * __Default value:__ `true`
-   */
-  domain?: boolean;
-
-  /**
-   * A boolean flag indicating if grid lines should be included as part of the axis
-   *
-   * __Default value:__ `true` for [continuous scales](https://vega.github.io/vega-lite/docs/scale.html#continuous) that are not binned; otherwise, `false`.
+   * An interpolation fraction indicating where, for `band` scales, axis ticks should be positioned. A value of `0` places ticks at the left edge of their bands. A value of `0.5` places ticks in the middle of their bands.
```
Comment (kanitw) [inline src/vega.schema.ts]: 0.5 is the default according to https://github.com/vega/vega-parser/blob/master/src/config.js.

File: src/vega.schema.ts [hunk 3]
```diff
@@ -739,6 +761,8 @@ export interface VgAxisConfig extends VgAxisBase {
   titleY?: number;
 }
 
+export interface VgAxisConfig extends VgAxisBase {}
```
Comment (kanitw) [inline src/vega.schema.ts]: If all properties of `VgAxisBase` are in `VgAxisConfig`, then we don't need to declare a separate `VgAxisBase`.

Comment (melissatdiamond) [reply src/vega.schema.ts]: I think `Axis` in `src/axis.ts` also extends `VgAxisBase` but has additional properties as well. Should I change `VgAxisBase` to `VgAxisConfig` and have `Axis` extend that?

Comment (kanitw) [reply src/vega.schema.ts]: Yep

File: src/vega.schema.ts [hunk 4]
```diff
@@ -501,145 +525,118 @@ export type VgBinding = VgCheckboxBinding | VgRadioBinding |
...
    */
-  labels?: boolean;
+  domainWidth?: number;
 
   /**
-   * Indicates if labels should be hidden if they exceed the axis range. If `false `(the default) no bounds overlap analysis is performed. If `true`, labels will be hidden if they exceed the axis range by more than 1 pixel. If this property is a number, it specifies the pixel tolerance: the maximum amount by which a label bounding box may exceed the axis range.
+   * Color of axis domain line.
    *
-   * __Default value:__ `false`.
+   * __Default value:__  `gray`.
```
Comment (kanitw) [inline src/vega.schema.ts]: We need to include quotes for strings (e.g., `"gray"`).  Please do the same for other properties.

**Pull request merged**